props/blaster.h

```diff
@@ -61,23 +45,74 @@ class Blaster : public PropBase {
   }
 
   bool auto_firing_ = false;
+  int shots_fired_ = 0;
+
+#ifdef BLASTER_SHOTS_UNTIL_EMPTY
+  const int max_shots_ = BLASTER_SHOTS_UNTIL_EMPTY;
+#else
+  const int max_shots_ = -1;
+#endif
 
   virtual void Fire() {
-    if (!SaberBase::IsOn()) return; // Don't allow firing when off.
+    if (max_shots_ != -1) {
+      if (shots_fired_ >= max_shots_) {
+        SaberBase::DoEmpty();
+        return;
+      }
+    }
 
     if (blaster_mode == MODE_AUTO) {
+      SelectAutoFirePair(); // Set up the autofire pairing if the font suits it.
       SaberBase::SetLockup(LOCKUP_AUTOFIRE);
       SaberBase::DoBeginLockup();
       auto_firing_ = true;
     } else {
-      SaberBase::DoBlast();
-#ifdef ENABLE_AUDIO
       if (blaster_mode == MODE_STUN) {
-        hybrid_font.PlayCommon(&SFX_stun);
+        SaberBase::DoStun();
       } else {
-        hybrid_font.PlayCommon(&SFX_blast);
+        SaberBase::DoFire();
       }
-#endif
+
+      shots_fired_++;
+    }
+  }
+
+  virtual void SelectAutoFirePair() {
+    if (!SFX_auto.files_found() || !SFX_blast.files_found()) return;
+
+    int autoCount = SFX_auto.files_found();
+    int blastCount = SFX_blast.files_found();
+    int pairSelection;
+
+    // If we don't have a matched pair of autos and blasts, then don't override the sequence to get a matched pair.
+    if (autoCount == blastCount) {
+        pairSelection = rand() % autoCount;
+        SFX_auto.Select(pairSelection);
+        SFX_blast.Select(pairSelection);
+    }
+  }
+
+  virtual void Reload() {
+    shots_fired_ = 0;
+    SaberBase::DoReload();
+    SaberBase::DoFull();
```

Comment:
This seems redundant.
Why not just have DoReload() play both sounds?

Reply:
I'm not entirely sure where this is supposed to happen but I wanted different effects for the different font files so they could be styled independently and still use the base approach of the SaberBase created functions.

Reply:
If we don't know what it's supposed to do, it's probably better to not trigger that effect at all for now.
Some fonts will use it, and if it's triggered at the wrong time, it will sound weird.

Reply:
Yeah ok. I'll pull the call to it but leave the capability for it there? So once we figure out how it is actually supposed to be used I can add the correct behavior in the blaster prop.

Reply:
sounds good to me